src/main/scala/com/github/nikalaikina/poehali/api/RestInterface.scala

```diff
@@ -41,43 +42,54 @@ trait RestApi extends HttpService { actor: Actor with AskSupport =>
 
 
   def routes: Route =
-    pathPrefix("flights") {
-      pathEnd {
-        get {
-          parameters('homeCities, 'cities, 'dateFrom, 'dateTo, 'daysFrom.as[Int], 'daysTo.as[Int], 'cost.as[Int], 'citiesCount.as[Int]) {
-            (homeCities, cities, dateFrom, dateTo, daysFrom, daysTo, cost, citiesCount) =>  {
-              respondWithMediaType(`application/json`) { (ctx: RequestContext) =>
-                val settings = new Trip(homeCities, cities, dateFrom, dateTo, daysFrom, daysTo, cost, citiesCount)
-                (logic(settings) ? GetRoutees)
-                  .mapTo[Routes]
-                  .map(r => r.routes.map(tr => JsonRoute(tr.flights)))
-                  .map { x => ctx.complete(Json.toJson(x).toString) }
-              }
+    respondWithHeader(RawHeader("Access-Control-Allow-Origin", "*")) {(
+      pathPrefix("flights") {
+        pathEnd {
+          get {
+            parameters('homeCities, 'cities, 'dateFrom, 'dateTo, 'daysFrom.as[Int], 'daysTo.as[Int], 'cost.as[Int], 'citiesCount.as[Int]) {
+              processFlightsRequest
+            }
+          } ~
+          post {
+            formField('homeCities, 'cities, 'dateFrom, 'dateTo, 'daysFrom.as[Int], 'daysTo.as[Int], 'cost.as[Int], 'citiesCount.as[Int]) {
```

Comment:
use [entity](http://spray.io/documentation/1.2.3/spray-routing/marshalling-directives/entity/#entity) because request is just a json, is not a form